compiler/rustc_const_eval/src/const_eval/mod.rs

```diff
@@ -71,8 +71,7 @@ pub fn tag_for_variant_provider<'tcx>(
     let (ty, variant_index) = key.value;
     assert!(ty.is_enum());
 
-    let ecx =
-        InterpCx::new(tcx, ty.default_span(tcx), key.typing_env, crate::const_eval::DummyMachine);
+    let ecx = InterpCx::new(tcx, DUMMY_SP, key.typing_env, crate::const_eval::DummyMachine);
```

Comment:
We don't need this span anyways, and even if we did, it would be nice to use a relevant def span since it's just an enum anyways.